src/panels/lovelace/editor/view-editor/hui-dialog-edit-view.ts

```diff
@@ -111,94 +118,155 @@ export class HuiDialogEditView extends LitElement {
     }
 
     let content;
-    switch (this._curTab) {
-      case "tab-settings":
-        content = html`
-          <hui-view-editor
-            .isNew=${this._params.viewIndex === undefined}
-            .hass=${this.hass}
-            .config=${this._config}
-            @view-config-changed=${this._viewConfigChanged}
-          ></hui-view-editor>
-        `;
-        break;
-      case "tab-badges":
-        content = html`
-          ${this._badges?.length
-            ? html`
-                ${VIEWS_NO_BADGE_SUPPORT.includes(this._type)
-                  ? html`
-                      <ha-alert alert-type="warning">
-                        ${this.hass!.localize(
-                          "ui.panel.lovelace.editor.edit_badges.view_no_badges"
-                        )}
-                      </ha-alert>
-                    `
-                  : ""}
-                <div class="preview-badges">
-                  ${this._badges.map(
-                    (badgeConfig) => html`
-                      <hui-badge-preview
-                        .hass=${this.hass}
-                        .config=${badgeConfig}
-                      ></hui-badge-preview>
-                    `
-                  )}
-                </div>
-              `
-            : ""}
-          <hui-entity-editor
-            .hass=${this.hass}
-            .entities=${this._badges}
-            @entities-changed=${this._badgesChanged}
-          ></hui-entity-editor>
-        `;
-        break;
-      case "tab-visibility":
-        content = html`
-          <hui-view-visibility-editor
-            .hass=${this.hass}
-            .config=${this._config}
-            @view-visibility-changed=${this._viewVisibilityChanged}
-          ></hui-view-visibility-editor>
-        `;
-        break;
-      case "tab-cards":
-        content = html` Cards `;
-        break;
+
+    if (this._yamlMode) {
+      const viewConfig = {
+        ...this._config,
+        badges: this._badges,
+      };
```

Comment:
You might want to do this in `updated` only on the switch from UI editor to YAML editor instead of in every render, as it is not used after the first time. With `HaYamlEditor.setValue`.